sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -555,18 +555,39 @@ ssize_t coap_build_reply(coap_pkt_t *pkt, unsigned code,
 ssize_t coap_build_hdr(coap_hdr_t *hdr, unsigned type, uint8_t *token, size_t token_len, unsigned code, uint16_t id)
 {
     assert(!(type & ~0x3));
-    assert(!(token_len & ~0x1f));
+
+    uint16_t tkl_ext;
+    uint8_t tkl_ext_len, tkl;
+
+    if (token_len > 268) {
+        tkl_ext_len = 2;
+        tkl_ext = htons(token_len  - 255 - 14);
```

Comment:
Why not `-269` as suggested by the RFC?
I would also love to see a small comment on both if-case stating their corresponding case in the rfc, e.g. `/* e. TKL case 14: using 16-bit usinged int in netw. byte order */`

Reply:
Hm we could just copy & paste that explanation into a comment. :speak_no_evil: